server/controllers/entities/lib/get_entity_type.coffee

```diff
@@ -2,23 +2,49 @@ __ = require('config').universalPath
 _ = __.require 'builders', 'utils'
 { types } =  __.require 'lib', 'wikidata/aliases'
 
-# Takes an entity wdt:P31 (instance of) claims array
+# Takes an entity claims
 # Returns a entity type string: work, edition, article, human, genre
-# If no type is found, try with the wdt:P279 (subclass of) claims array
-# (used for Wikidata entities only, as all inv entities have a known P31)
-module.exports = (wdtP31Array, wdtP279Array)->
-  unless wdtP31Array? then return
+# If no type is found from wdt:P31 (instance of), try with wdt:P279 (subclass of)
+# (useful for Wikidata entities only, as all inv entities have a known P31)
+module.exports = (claims)->
+  type = guessType claims, 'wdt:P31'
+  if type? then return type
 
-  for value in wdtP31Array
-    type = types[value]
-    # return as soon as we get a type
-    if type? then return type
+  type = guessType claims, 'wdt:P279'
+  if type? then return type
 
-  unless wdtP279Array? then return
+  return
 
-  for value in wdtP279Array
-    type = types[value]
-    # return as soon as we get a type
-    if type? then return type
+guessType = (claims, property)->
+  propertyClaims = claims[property]
+  unless propertyClaims? then return
 
-  return
+  propertyTypes = getTypes propertyClaims
+
+  switch propertyTypes.length
+    when 0 then return
+    when 1
+      # Best case: deduce type from a unique P31 value
+      type = propertyTypes[0]
+      if type is 'edition'
+        # Wikidata contributors may have set an item to be an edition, while it's actually a mix
+        # of work and edition attributes. In absence of a clear edition shape, we consider it to be
+        # a work
+        unless claims['wdt:P629']? then return 'work'
+      return type
+    else
+      # Case with multiple P31 values: try to guess the most credible type
+      isWork = 'work' in propertyTypes
+      isEdition = 'edition' in propertyTypes
+      isSerie = 'serie' in propertyTypes
+
+      # If it has 'edition of' claims, consider that it's an edition
+      if isEdition and claims['wdt:P629']? then return 'edition'
```

Comment:
We might consider returning a `undefined` value as no wikidata edition are to be interpreted by inventaire has an edition. Inventaire would then consider this entity as a topic, not an edition, avoiding for example the possibility of adding it to a user's inventaire.